src/Hooks.ts

```diff
@@ -17,18 +17,69 @@ import {
   EventType,
 } from 'react-navigation';
 
-export function useNavigation<S>(): NavigationScreenProp<S & NavigationRoute> {
+function useNavigationSafe<S>(): NavigationScreenProp<S & NavigationRoute> {
   const navigation = useContext(NavigationContext) as any; // TODO typing?
+
   if (!navigation) {
     throw new Error(
       "react-navigation hooks require a navigation context but it couldn't be found. " +
         "Make sure you didn't forget to create and render the react-navigation app container. " +
         'If you need to access an optional navigation object, you can useContext(NavigationContext), which may return'
     );
   }
+
   return navigation;
 }
 
+function useStableCallback(cb: (...args: any) => any) {
```

Comment:
in terms of typing this is not good enough, all wrapped actions will be any[] => any

Reply:
The public api will have the same types as now, the `(...args: any) => any` will remain private. It's bad, but not so bad.